zokrates_ast/src/ir/serialize.rs

```diff
@@ -58,33 +63,195 @@ impl<
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum SectionType {
+    Arguments = 1,
```

Comment:
Because this number does not represent an index, but rather an identifier of a section type. I took a similar approach to this https://github.com/iden3/r1csfile/blob/master/doc/r1cs_bin_format.md#sectiontype